From cb430b93c85f89974147abb33ec46cdae5bdfe23 Mon Sep 17 00:00:00 2001
From: Vishveshwar Bhat <vishveshwar.bhat@ti.com>
Date: Wed, 29 Dec 2010 11:40:56 +0530
Subject: [PATCH 1/2] hcitool: Bring up device before sending commands over socket

The Texas specific initialization code sends HCI commands over the
bluetooth socket, but does not bring up the device. This gives these
errors when running "hciattach /dev/ttyUSB0 texas":

Found a Texas Instruments' chip!
Firmware file : /lib/firmware/TIInit_XX.Y.ZZ.bts
Loaded BTS script version 1
Cannot send hci command to socket: Network is down
Can't initialize device: Network is down

This patch is by Anderson Lizardo and taken from the following bluez
linux-bluetooth post http://marc.info/?l=linux-bluetooth&m=128932744131672&w=2

Signed-off-by: Vishveshwar Bhat <vishveshwar.bhat@ti.com>
---
 tools/hciattach_ti.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/tools/hciattach_ti.c b/tools/hciattach_ti.c
index e169f89..f067c5d 100644
--- a/tools/hciattach_ti.c
+++ b/tools/hciattach_ti.c
@@ -514,6 +514,13 @@ int texas_post(int fd, struct termios *ti)
 		return -1;
 	}
 
+	if (ioctl(dd, HCIDEVUP, dev_id) < 0 && errno != EALREADY) {
+		fprintf(stderr, "Can't init device hci%d: %s (%d)", dev_id,
+							strerror(errno), errno);
+		hci_close_dev(dd);
+		return -1;
+	}
+
 	ret = brf_do_script(dd, ti, NULL);
 
 	hci_close_dev(dd);
-- 
1.7.0.4

